pipetools/main.py

```diff
@@ -49,15 +51,32 @@ def __or__(self, next_func):
         # this is needed because of evaluation order
         pipe_in_a_pipe = isinstance(next_func, Pipe) and next_func.func is None
         new_cls = type(next_func) if pipe_in_a_pipe else None
-        next = None if pipe_in_a_pipe else prepare_function_for_pipe(next_func)
-        return self.bind(self.func, next, new_cls)
+        if pipe_in_a_pipe:
+            next = None
+            return self.bind(self.func, next, new_cls)
+        if isinstance(next_func, (Callable, str))\
+                or (isinstance(next_func, tuple) and isinstance(next_func[0], Callable)):
+            next = prepare_function_for_pipe(next_func)
+            return self.bind(self.func, next, new_cls)
+        current_func = prepare_function_for_pipe((self.func, next_func))
+        next = None
+        return self.bind(current_func, next, new_cls)
 
     def __ror__(self, prev_func):
         return self.bind(prepare_function_for_pipe(prev_func), self.func)
 
     def __lt__(self, thing):
         return self.func(thing) if self.func else thing
 
+    def __rrshift__(self, thing):
```

Comment:
I don't think we really want this. Because of operator precedence it requires to put parentheses around the pipe and it will probably confuse people when the forget to do it. Also adding another way to feed the pipe might be a bit too much :) I'd like to keep it as simple as possible.